# TensorFlow2.0-beta1  


In [1]:
from __future__ import absolute_import, division, print_function, unicode_literals

import tensorflow as tf

import tensorflow

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

ready for MNIST dataset
* divide it to train set and test set
* convert value from `0~255` to `0~1`

In [6]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

make model with `tf.keras.Sequential` and select compile options

In [7]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

W0726 00:28:19.680713 11144 deprecation.py:323] From C:\Users\82103\Anaconda3\envs\cuda\lib\site-packages\tensorflow\python\ops\math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 13s 214us/sample - loss: 0.2941 - accuracy: 0.9160
Epoch 2/5
60000/60000 [==============================] - 7s 118us/sample - loss: 0.1411 - accuracy: 0.9582
Epoch 3/5
60000/60000 [==============================] - 7s 116us/sample - loss: 0.1080 - accuracy: 0.9669
Epoch 4/5
60000/60000 [==============================] - 7s 114us/sample - loss: 0.0874 - accuracy: 0.9733
Epoch 5/5
10000/10000 [==============================] - 1s 65us/sample - loss: 0.0740 - accuracy: 0.9773


[0.07403070160984061, 0.9773]

train model and evaluate(ingnore warning)

total loss = 0.07 (7%)  
total accuracy : 0.9773